File: backend/Models/TaxCompliance.cs
namespace KasseAPI_Final.Models;

public class ComplianceIssue
{
    /// <summary>Critical | Warning | Info</summary>
    public string Severity { get; set; } = "Warning";

    /// <summary>Stable machine code for FA i18n (e.g. MISSING_TAX_GROUP).</summary>
    public string Code { get; set; } = string.Empty;

    public string Message { get; set; } = string.Empty;

    public string Action { get; set; } = string.Empty;

    public int AffectedCount { get; set; }

    public IReadOnlyList<Guid> SampleProductIds { get; set; } = [];
}

public class ComplianceReport
{
    public bool IsCompliant { get; set; }

    public List<ComplianceIssue> Issues { get; set; } = [];

    public int TotalProducts { get; set; }

    public int CompliantProducts { get; set; }

    public int NonCompliantProducts { get; set; }

    public DateTime CheckedAtUtc { get; set; } = DateTime.UtcNow;
}

File: frontend-admin/src/features/tax/api/taxCompliance.ts
import { customInstance } from '@/lib/axios';

export type ComplianceIssue = {
  severity: string;
  code: string;
  message: string;
  action: string;
  affectedCount: number;
  sampleProductIds?: string[];
};

export type ComplianceReport = {
  isCompliant: boolean;
  issues: ComplianceIssue[];
  totalProducts: number;
  compliantProducts: number;
  nonCompliantProducts: number;
  checkedAtUtc: string;
};

export const taxComplianceQueryKey = ['tax-compliance'] as const;

export async function getTaxComplianceReport(): Promise<ComplianceReport> {
  return customInstance<ComplianceReport>({
    url: '/api/admin/tax-compliance',
    method: 'GET',
  });
}
